src/Compilers/VisualBasic/Test/Semantic/Diagnostics/GetDiagnosticsTests.vb

```diff
@@ -224,6 +224,189 @@ End Namespace
+            Dim dict1 = New Dictionary(Of String, Object)
+            dict1.Add("1", Nothing)
+            Dim dict2 = New Dictionary(Of String, Object)
+            dict2.Add("2", Nothing)
+
+            Dim parseOptions1 = New VisualBasicParseOptions().WithPreprocessorSymbols(dict1)
+            Dim parseOptions2 = New VisualBasicParseOptions().WithPreprocessorSymbols(dict2)
+
+            Dim syntaxTree1 = Parse(String.Empty, options:=parseOptions1)
+            Dim syntaxTree2 = Parse(String.Empty, options:=parseOptions2)
+
+            Dim options = New VisualBasicCompilationOptions(OutputKind.DynamicallyLinkedLibrary, parseOptions:=parseOptions1)
+            Dim compilation = CreateCompilationWithMscorlibAndVBRuntime({syntaxTree1, syntaxTree2}, options:=options)
+            Dim diagnostics = compilation.GetDiagnostics()
+
+            CompilationUtils.AssertTheseDiagnostics(diagnostics,
+<errors>
+BC31030: Conditional compilation constant '1' is not valid: Identifier expected.
+BC31030: Conditional compilation constant '2' is not valid: Identifier expected.
+
+~
+</errors>)
+
+            Assert.Equal(diagnostics(0).Arguments, {"Identifier expected.", "2"})
+            Assert.True(diagnostics(0).Location.SourceTree.Equals(syntaxTree2)) ' Syntax tree parse options are reported in CompilationStage.Parse
+
+            Assert.Equal(diagnostics(1).Arguments, {"Identifier expected.", "1"})
+            Assert.Null(diagnostics(1).Location.SourceTree) ' Compilation parse options are reported in CompilationStage.Declare
+        End Sub
+
+        <Fact>
+        Public Sub DiagnositcsInCompilationOptionsParseOptionsAreReportedSeparately()
+            Dim dict1 = New Dictionary(Of String, Object)
+            dict1.Add("1", Nothing)
+            Dim dict2 = New Dictionary(Of String, Object)
+            dict2.Add("2", Nothing)
+
+            Dim parseOptions1 = New VisualBasicParseOptions().WithPreprocessorSymbols(dict1)
+            Dim parseOptions2 = New VisualBasicParseOptions().WithPreprocessorSymbols(dict2)
+
+            Dim options = New VisualBasicCompilationOptions(OutputKind.DynamicallyLinkedLibrary, parseOptions:=parseOptions1)
+
+            CompilationUtils.AssertTheseDiagnostics(options.Errors,
+<errors>
+BC31030: Conditional compilation constant '1' is not valid: Identifier expected.
+</errors>)
+
+            Dim syntaxTree = Parse(String.Empty, options:=parseOptions2)
+            Dim compilation = CreateCompilationWithMscorlibAndVBRuntime({syntaxTree}, options:=options)
+            Dim diagnostics = compilation.GetDiagnostics()
+
+            CompilationUtils.AssertTheseDiagnostics(diagnostics,
+<errors>
+BC31030: Conditional compilation constant '1' is not valid: Identifier expected.
+BC31030: Conditional compilation constant '2' is not valid: Identifier expected.
+
+~
+</errors>)
+
+            Assert.Equal(diagnostics(0).Arguments, {"Identifier expected.", "2"})
```

Comment:
CompilationUtils sorts errors before comparison. Added a comment and an assert for the correct order.